src/Planners/src/UnicyclePlanner.cpp

```diff
@@ -0,0 +1,511 @@
+                           pImpl->gait.resetStartingFootIfStill))
+    {
+        log()->info("{} Using default gait_reset_starting_foot_if_still={}.",
+                    logPrefix,
+                    pImpl->gait.resetStartingFootIfStill);
+    }
+
+    pImpl->planner->startWithLeft(pImpl->gait.startWithLeft);
+    pImpl->planner->addTerminalStep(pImpl->gait.terminalStep);
+    pImpl->planner->resetStartingFootIfStill(pImpl->gait.resetStartingFootIfStill);
+
+    // =============================
+    // UnicyclePlanner configuration
+    // =============================
+
+    if (!okPlanner)
+    {
+        log()->error("{} Failed to configure UnicyclePlanner.", logPrefix);
+        return false;
+    }
+
+    return true;
+}
+
+const Planners::UnicyclePlannerState& Planners::UnicyclePlanner::getOutput() const
+{
+    constexpr auto logPrefix = "[UnicyclePlanner::getOutput]";
+
+    pImpl->state = UnicyclePlannerState();
+
+    if (!this->isOutputValid())
+    {
+        return pImpl->state;
+    }
+
+    auto convertStep
+        = [&](const ::Step& step, const ::Step& nextStep) -> const Contacts::PlannedContact {
+        Contacts::PlannedContact contact;
+        contact.name = step.footName;
+        contact.activationTime = step.impactTime;
+
+        contact.pose.quat(Eigen::AngleAxisd(step.angle, Eigen::Vector3d::UnitZ()));
+        contact.pose.translation({step.position[0], step.position[1], 0.0});
+
+        // Note: we use the impact time to compute the duration of the step.
+        //       If the step is the last one, a default 5 seconds duration is used.
+        const double duration = nextStep.impactTime - step.impactTime;
+        assert(duration >= 0);
+        contact.deactivationTime
+            = duration > 0.0 ? contact.activationTime + duration * pImpl->gait.stancePhaseRatio
+                             : contact.activationTime + 5.0;
+
+        return contact;
+    };
+
+    auto convertSteps
+        = [&convertStep](decltype(
+              ::FootPrint().getSteps())& steps) -> std::optional<const Contacts::ContactList> {
+        Contacts::ContactList list;
+
+        for (std::size_t i = 0; i < steps.size(); ++i)
+        {
+            // Last step
+            if (i == steps.size())
+            {
+                if (!list.addContact(convertStep(steps[i], steps[i])))
+                {
+                    return std::nullopt;
+                }
+                break;
+            }
+
+            // All steps but last
+            if (!list.addContact(convertStep(steps[i], steps[i + 1])))
+            {
+                return std::nullopt;
+            }
+        }
+
+        return list;
+    };
+
+    const auto& left = convertSteps(pImpl->left->getSteps());
+    const auto& right = convertSteps(pImpl->right->getSteps());
+
+    if (!left || !right)
+    {
+        log()->error("{} Failed to convert unicycle steps.", logPrefix);
+        return pImpl->state;
+    }
```

Comment:
To be honest, I am not very convinced in doing a lot of logic in ``getOutput``. ``advance`` is where the logic should be. ``getOuput`` should be a simple getter. There could be a case where ``isOutputValid`` returns true, but the logic in here fails returning an invalid output.

Reply:
I'll be moving the contact conversion logic to the `advance` if it's ok for you.

Reply:
Done in 777cb1d